terratest/test/helper.go

```diff
@@ -19,25 +19,51 @@ package test
 
 import (
 	"fmt"
-	"strings"
-	"sort"
 	"testing"
-	"github.com/gruntwork-io/terratest/modules/shell"
+	"time"
+
+	"github.com/miekg/dns"
 )
 
-func Dig(t *testing.T, dnsServer string, dnsPort int, dnsName string) ([]string, error) {
-        port := fmt.Sprintf("-p%v", dnsPort)
-        dnsServer = fmt.Sprintf("@%s", dnsServer)
+const (
+	DefaultTimeout time.Duration = 5 * time.Second
+)
 
-        digApp := shell.Command{
-                Command: "dig",
-                Args:    []string{port, dnsServer, dnsName, "+short"},
-        }
+func queryDNS(dnsServer string, dnsPort int, dnsName string, dnsType uint16) (*dns.Msg, error) {
+	dnsName = fmt.Sprintf("%s.", dnsName)
+	m := &dns.Msg{
+		MsgHdr: dns.MsgHdr{
+			RecursionDesired: true,
+		},
+		Question: make([]dns.Question, 1),
+	}
+	c := &dns.Client{
+		ReadTimeout: DefaultTimeout,
+	}
+	c.Net = "udp4"
+	m.SetQuestion(dnsName, dnsType)
+	r, _, err := c.Exchange(m, fmt.Sprintf("%s:%d", dnsServer, dnsPort))
+	return r, err
+}
+
+func DigMsg(t *testing.T, dnsServer string, dnsPort int, dnsName string, dnsType uint16) (*dns.Msg, error) {
+	return queryDNS(dnsServer, dnsPort, dnsName, dnsType)
+}
 
-        digAppOut := shell.RunCommandAndGetOutput(t, digApp)
-        digAppSlice := strings.Split(digAppOut, "\n")
+func DigIPs(t *testing.T, dnsServer string, dnsPort int, dnsName string, dnsType uint16) ([]string, error) {
+	var result []string
+	r, err := queryDNS(dnsServer, dnsPort, dnsName, dnsType)
 
-        sort.Strings(digAppSlice)
+	if err != nil {
+		return nil, err
+	}
 
-        return digAppSlice, nil
+	for _, record := range r.Answer {
+		if e, ok := record.(*dns.A); ok {
+			if e.A != nil {
+				result = append(result, e.A.String())
+			}
```

Comment:
what if `*e.A == nil`  ? Shouldn't be an error returned ?

Reply:
@kuritka I have to do a research on this. Wether standard allows A record without address